### PR TITLE
Remove support for storage v6.

### PR DESCRIPTION
**Pull request by przydatek**
Remove the support for storage layout v6, as the migration to v7 succeeded in [release-2023-05-30](https://github.com/dfinity/internet-identity/tree/release-2023-05-30) ([proposal 122633](https://nns.ic0.app/proposal/?u=qoctq-giaaa-aaaaa-aaaea-cai&proposal=122633)). 
All the code and tests specific to v6 are removed, except for the flag `InternetIdentityInit.migrate_storage_to_memory_manager`, which is needed for upgrade/downgrade tests.  The flag will be removed in a follow-up PR, for a subsequent release.
Some tests not specific to v6 are adopted to work with v7. 

(GIX-1591)

### CODE REVIEW

**Review by nmattia — COMMENTED** [review]
Thanks! couple of questions

**Review by przydatek — COMMENTED** [review]
Thanks, PTAL.

File: src/internet_identity/tests/integration/http.rs [hunk 1]
```diff
@@ -277,16 +277,17 @@ fn metrics_stable_memory_pages_should_increase_with_more_users() -> Result<(), C
 
     let metrics = get_metrics(&env, canister_id);
     let (stable_memory_pages, _) = parse_metric(&metrics, "internet_identity_stable_memory_pages");
-    // empty II has some metadata in stable memory which requires at least one page
-    assert_eq!(stable_memory_pages, 1f64);
+    // empty II has some metadata in stable memory which requires at two pages:
+    // one page for the header, and one for the memory manager.
+    assert_eq!(stable_memory_pages, 2f64);
```
[the remark this answers is not in the document]
Comment (nmattia) [reply src/internet_identity/tests/integration/http.rs:282]: Ah my bad I wasn't clear, I meant that it's not clear to me why the number of pages changes even though there is not actual change in memory layout in this PR (AFAICT)

**Review by przydatek — COMMENTED** [review]
Thanks, PTAL.

**Review by nmattia — APPROVED** [review]
Not familiar enough with the code for a real review, but LGTM to unblock!

**Pull request merged**